### PR TITLE
Steps with neg sigma

### PR DESCRIPTION
**Pull request by newville**
#### Description

This allows the `step` lineshape and `StepModel` to have values of `sigma<0` to signify a step down (from 1 to 0).

See discussion at: https://groups.google.com/g/lmfit-py/c/aBBzQG0uDK4

###### Type of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring / maintenance
- [ ] Documentation / examples

###### Tested on

Python: 3.12.5 | packaged by conda-forge | (main, Aug  8 2024, 18:31:54) [Clang 16.0.6 ]

lmfit: 1.3.2.post9+g776e14d7.d20241007, scipy: 1.14.0, numpy: 1.26.4, asteval: 1.0.5, uncertainties: 3.2.2.post7+gce16269
###### Verification 
Have you

- [x] included docstrings that follow PEP 257?

- [x] referenced existing Issue and/or provided relevant link to mailing list?

- [x] verified that existing tests pass locally?

- [x] verified that the documentation builds locally?

- [ ] squashed/minimized your commits and written descriptive commit messages?

- [x] added or updated existing tests to cover the changes?
- [x] updated the documentation and/or added an entry to the release notes (doc/whatsnew.rst)?
- [ ] added an example?

### CODE REVIEW

File: lmfit/lineshapes.py [hunk 1]
```diff
@@ -415,8 +415,8 @@ def thermal_distribution(x, amplitude=1.0, center=0.0, kt=1.0, form='bose'):
 def step(x, amplitude=1.0, center=0.0, sigma=1.0, form='linear'):
     """Return a step function.
 
-    Starts at 0.0, ends at `amplitude`, with half-max at `center`, and
-    rising with `form`:
+    Starts at 0.0, ends at `sign(sigma)*amplitude`, has a half-max at
+    `center`, rsing or falling with `form`:
```
Comment (reneeotten) [inline lmfit/lineshapes.py:419]: typo: rsing --> rising

File: lmfit/models.py [hunk 2]
```diff
@@ -1525,8 +1525,9 @@ class StepModel(Model):
       https://en.wikipedia.org/wiki/Logistic_function)
 
     The step function starts with a value 0 and ends with a value of
-    :math:`A` rising to :math:`A/2` at :math:`\mu`, with :math:`\sigma`
-    setting the characteristic width. The functional forms are defined as:
+    :math:`\tt{sign}(\sigma)A` rising or falling to :math:`A/2` at :math:`\mu`,
+    with :math:`\sigma` setting the characteristic width and the sign up the step.
```
Comment (reneeotten) [inline lmfit/models.py:1529]: "sign up" is that what you meant?

**Review by reneeotten — COMMENTED** [review]
thanks @newville a few minor typos perhaps but other than that looks good to me. Perhaps just squash the commits?

File: lmfit/lineshapes.py (hunk 1 above)
Comment (newville) [reply lmfit/lineshapes.py]: thanks, fixed.

File: lmfit/models.py (hunk 2 above)
Comment (newville) [reply lmfit/models.py]: thanks, fixed.

**Comment by newville** [comment]
@reneeotten OK to merge?

**Comment by reneeotten** [comment]
> @reneeotten OK to merge?

Yep, please squash and update the commit message

**Pull request merged**